File: 0396-rotate-function/0396-rotate-function.cpp
class Solution {
public:
    int maxRotateFunction(vector<int>& nums) {
        int n = nums.size();
        int res = 0;
        int last = 0;
        int tot = 0;
        for(int i = 0; i < n; i++)
        {
            last += (i * nums[i]);
            tot += nums[i];
        }
        res = last;
        for(int i = 0; i < n - 1; i++)
        {
            last += tot - n * nums[n - i - 1];
            res = max(res, last);
        }
        return res;
    }
};

File: 0396-rotate-function/0396-rotate-function.py
class Solution:
    def maxRotateFunction(self, nums: List[int]) -> int:
        n = len(nums)
        last = 0
        tot = 0
        for i in range(n):
            last += (i * nums[i])
            tot += nums[i]
        res = last
        for i in range(n - 1):
            last = last + tot - n * nums[n - i - 1]
            res = max(res, last)
        return res